### PR TITLE
[yang]: Add yang model for MACsec

### PR DESCRIPTION
**Pull request by Pterosaur**
#### Why I did it
Add Yang model to constrain the configuration of MACsec

#### How I did it
Add Yang model for MACsec configuration and its testcases

#### How to verify it

Check Azp status

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Link to config_db schema for YANG module changes
https://github.com/Azure/SONiC/blob/master/doc/macsec/MACsec_hld.md#31-config-db

### CODE REVIEW

**Comment by ganglyu** [comment]
Build failed, please add yang model to setup.py
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/setup.py#L80-#L198

**Comment by ganglyu** [comment]
Please update tests/files/sample_config_db.json

**Comment by Pterosaur** [comment]
> Please update tests/files/sample_config_db.json

Done

File: src/sonic-yang-models/setup.py [hunk 1]
```diff
@@ -143,7 +143,8 @@ def run(self):
                          './yang-models/sonic-tc-queue-map.yang',
                          './yang-models/sonic-pfc-priority-queue-map.yang',
                          './yang-models/sonic-pfc-priority-priority-group-map.yang',
-                         './yang-models/sonic-port-qos-map.yang']),
+                         './yang-models/sonic-port-qos-map.yang',
+                         './yang-models/sonic-macsec.yang',]),
```
Comment (ganglyu) [inline src/sonic-yang-models/setup.py:147]: Would you please remove the ','

File: src/sonic-yang-models/yang-models/sonic-macsec.yang [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                leaf policy {
+                    type string {
+                        pattern "integrity_only|security";
+                    }
+                    default "security";
+                }
+
+                leaf enable_replay_protect {
+                    type string {
+                        pattern "true|false";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-macsec.yang:87]: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-feature.yang#L58
Can you use boolean_type?

**Review by ganglyu — APPROVED** [review]

**Pull request merged**